### DISCUSSION TITLE
Pre-release schedule

### DISCUSSION THREAD

**Discussion by Xitric** [testcontainers/testcontainers-dotnet#513] [General]
I seem to remember that the schedule for releasing pre-releases (beta versions) of this package was quite frequent in version 1.x, but since the change to version 2.x the pre-releases seem to have stagnated, especially considering all the great new feature development that has been going on in the past month.

Has something changed in the pre-release schedule, or am I simply imagining things?

The reason I ask is that we would like to run internal tests of some of the new features that have been added recently, while still using Nuget package references as opposed to a project reference of a local repository clone.

**Comment by HofmeisterAn** [comment]
> Has something changed in the pre-release schedule, or am I simply imagining things?

No, you're right 😅.

With the switch to GitHub Actions, I changed the deployment strategy a bit. Previously, each commit to develop published a new snapshot version. Event if it doesn't contain any value for the devs. Now, we can specific select the version we would like to publish (manual).

I can adjust the release strategy whatever is best for the community. What do you think would be a good approach? At least once a month? Each pull request that adds value for the devs?

BTW, if you need a snapshot version, I can deploy one.

**Reply by Xitric** [reply]
I recognize that this is perhaps not easily automated. A regular release interval could work, but I think that once a month would be too rare.

Perhaps when people open a pull request, they could declare if they would like for a pre-release to be deployed upon merging? Whether you handle it manually or automatically, I will let you decide 😉 We should of course also ensure that we don't end up releasing multiple pre-releases in a day!
